File: public/code/drafting-emails-via-the-api/5.py
import json

import requests

# You can find this key in your API requests page:
# https://buttondown.com/requests
BUTTONDOWN_API_KEY = "your-api-key-here"
EMAIL_ID = "your-email-id-here"

headers = {"Authorization": f"Token #{BUTTONDOWN_API_KEY}"}
data = {
    "status": "scheduled",
    "publish_date": "2027-08-24T14:15:22Z",
}
base_url = "https://api.buttondown.email"
endpoint = f"/v1/emails/{EMAIL_ID}"
response = requests.patch(base_url + endpoint, headers=headers, data=json.dumps(data))
print(response)

File: public/code/drafting-emails-via-the-api/5.rb
require "HTTParty"
require "json"

# You can find this key in your API requests page:
# https://buttondown.com/requests
BUTTONDOWN_API_KEY = "your-api-key-here"

response = HTTParty.patch(
  "https://api.buttondown.email/v1/emails/#{EMAIL_ID}",
  body: {
    status: "scheduled",
    publish_date: "2027-08-24T14:15:22Z"
  }.to_json,
  headers: {
    "Authorization" => "Token #{BUTTONDOWN_API_KEY}",
    "Content-Type" => "application/json"
  }
)
puts(JSON.parse(response.body))
print(response)
